### ISSUE TITLE
Does sequence parallel need all-reduce for gradients before optimizer.step()?

### ISSUE THREAD

**Issue by zxmeng98** [microsoft/Megatron-DeepSpeed#266]
Hi, 
I am reading the implementation code of sequence parallelism recently and didn't find the all-reduce of gradients after backward finishes. Because the sequence is split to sub-sequences before fed into network, I wonder should there be an all-reduce for gradients before optimizer.step()? Similar to data parallelism?

 
![dcf7a9afd4e5e0fdf6b759a6f490451](https://github.com/microsoft/Megatron-DeepSpeed/assets/96046956/363103ae-5bbb-45a9-9310-5e6b324b577a)
